include/git2/transport.h

```diff
@@ -105,6 +105,15 @@ typedef enum {
 	 * it will ask via this credential type.
 	 */
 	GIT_CREDTYPE_USERNAME = (1u << 5),
+
+#ifdef GIT_SSH_MEMORY_CREDENTIALS
```

Comment:
We shouldn't ifdef this. The value must be available regardless of whether the current build of the library supports this.